libraries/ms-common/inc/soft_timer.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "status.h"
+
+#define SOFT_TIMER_MAX_TIMERS 10
+
+typedef uint8_t SoftTimerID;
+
+typedef void (*SoftTimerCallback)(SoftTimerID timer_id, void* context);
+
+// Initializes a set of software timers. Clock speed should be in MHz. Subsequent calls will do
+// nothing. The clock speed is that of the external PLL crystal.
+void soft_timer_init(void);
+
+// Adds a software timer. The provided duration is the number of microseconds before running and the
+// callback is the process to run once the time has expired. The timer_id is set to the id of the
+// timer that will run the callback.
+StatusCode soft_timer_start(uint32_t duration_us, SoftTimerCallback callback, void* context,
```

Comment:
Might be nice to have convenience functions/macros for millisecond and second timers? Also, I assume periodic timers will just need to start a new timer in the timer callback?